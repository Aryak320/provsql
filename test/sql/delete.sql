\set ECHO none
\pset format unaligned
SET search_path TO provsql_test,provsql;

CREATE TABLE t(id INT PRIMARY KEY);

INSERT INTO t (id) 
VALUES (1), (2), (3), (4), (5);

SELECT add_provenance('t');

-- Test 1: single row deletion
SELECT create_provenance_mapping('t_id', 't', 'id');
SELECT COUNT(*) FROM t_id;
SELECT probability_evaluate(provenance) FROM t_id WHERE value = 1;

DROP TABLE t_id;
DELETE FROM t WHERE id = 1;
SELECT create_provenance_mapping('t_id', 't', 'id');
SELECT COUNT(*) FROM t_id;
SELECT probability_evaluate(provenance) FROM t_id WHERE value = 1;

-- Test 2: multiple rows deletion
DROP TABLE t_id;
SELECT create_provenance_mapping('t_id', 't', 'id');
SELECT COUNT(*) FROM t_id;
SELECT probability_evaluate(provenance) FROM t_id WHERE value = 1;

DROP TABLE t_id;
DELETE FROM t WHERE id >= 2 AND id <= 4;
SELECT create_provenance_mapping('t_id', 't', 'id');
SELECT COUNT(*) FROM t_id;
SELECT probability_evaluate(provenance) FROM t_id WHERE value = 1;

-- Test 3: delete token tracking/ logging into delete_provenance table
SELECT COUNT(*) FROM delete_provenance;

