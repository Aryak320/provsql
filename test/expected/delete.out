Output format is unaligned.
SET
CREATE TABLE
INSERT 0 5
add_provenance

(1 row)
create_provenance_mapping

(1 row)
count
5
(1 row)
probability_evaluate
1
(1 row)
DELETE 1
DROP TABLE
DELETE 1
create_provenance_mapping

(1 row)
count
5
(1 row)
probability_evaluate
0
(1 row)
DROP TABLE
create_provenance_mapping

(1 row)
count
5
(1 row)
probability_evaluate
0
(1 row)
DELETE 1
DROP TABLE
DELETE 3
create_provenance_mapping

(1 row)
count
5
(1 row)
probability_evaluate
0
(1 row)
count
7
(1 row)